Automated Smoketest 
Sha: 96b9191d09a6e31a6275ffb31ddc6e5464e026c2
Build: failed
Number of Error(s): 6
  Policy CMP0046 is not set: Error on non-existent dependency in
  Policy CMP0046 is not set: Error on non-existent dependency in
/mnt/disk1/home/vamosax/smoketest/PR-8889/HPCC-Platform/rtl/eclrtl/eclregex.cpp:347:54: error: \xe2\x80\x98nullptr\xe2\x80\x99 was not declared in this scope
make[2]: *** [rtl/eclrtl/CMakeFiles/eclrtl.dir/eclregex.cpp.o] Error 1
make[1]: *** [rtl/eclrtl/CMakeFiles/eclrtl.dir/all] Error 2
make: *** [all] Error 2

--- a/ecl/regress/regex4.ecl
+++ b/ecl/regress/regex4.ecl
@@ -1,0 +1,1 @@
+REGEXREPLACE(u'a[', 'b', 'c');

--- a/ecl/regress/regex5.ecl
+++ b/ecl/regress/regex5.ecl
@@ -1,0 +1,1 @@
+REGEXREPLACE('a[', 'b', 'c');

--- a/rtl/eclrtl/eclregex.cpp
+++ b/rtl/eclrtl/eclregex.cpp
@@ -344,7 +344,7 @@ public:
         else
             pattern = RegexPattern::compile(_UregExp, UREGEX_CASE_INSENSITIVE, uperr, uerr);
 
-        matcher = pattern->matcher(uerr);
+        matcher = pattern ? pattern->matcher(uerr) : NULL;
         if (U_FAILURE(uerr))
         {
             char * expAscii;
@@ -357,8 +357,8 @@ public:
             rtlFree(expAscii);
             delete matcher;
             delete pattern;
-            matcher = 0;
-            pattern = 0;
+            matcher = NULL;
+            pattern = NULL;
             rtlFail(0, msg.c_str());  //throws
         }
     }

